pkg/abstract/slices.go

```diff
@@ -75,6 +75,19 @@ func SliceMap[Domain, Range any](slice []Domain, fn func(Domain) Range) []Range
 	return ans
 }
 
+func SliceToMap[Domain, Range any, Key comparable](slice []Domain, keyFn func(Domain) Key,
```

Comment:
Perhaps name this `SliceToPrimitiveMap`, for consistency with the other names and anticipating the day when we have a general `Map` abstraction or simply not presuming that built-in maps are the only kind of map.